### ISSUE TITLE
Stop-loss on exchange issue?

### ISSUE THREAD

**Issue by Fredrik81** [freqtrade/freqtrade#3046]
Describe your environment

  * Operating system: Ubuntu 19.10
  * Python Version: 3.7.5
  * CCXT version: ccxt==1.23.30
  * Branch: Develop
  * Last Commit ID: 77944175e27bcd7851a83863f5df15f3577d61de
  
## Step 3: Describe the problem:
It looks like the calculation of new stop-loss is wrong sometimes.
Example:
```
2020-03-09 07:17:46,231 - freqtrade.freqtradebot - INFO - Trailing stoploss: cancelling current stoploss on exchange (id:{56124851}) in order to add another one ...
2020-03-09 07:17:47,742 - freqtrade.freqtradebot - ERROR - Unable to place a stoploss order on exchange. Could not create stop_loss_limit sell order on market BCH/BTC. Tried to sell amount 0.01 at rate 0.03451. Message: binance Order would trigger immediately.
2020-03-09 07:17:47,743 - freqtrade.freqtradebot - WARNING - Selling the trade forcefully
2020-03-09 07:17:48,541 - freqtrade.wallets - INFO - Wallets synced.
2020-03-09 07:17:49,342 - freqtrade.persistence - INFO - Updating trade (id=67) ...
2020-03-09 07:17:49,342 - freqtrade.persistence - INFO - Marking Trade(id=67, pair=BCH/BTC, amount=0.01000000, open_rate=0.03463000, open_since=closed) as closed as the trade is fulfilled and found no open orders for it.
2020-03-09 07:17:49,342 - freqtrade.persistence - INFO - MARKET_SELL has been fulfilled for Trade(id=67, pair=BCH/BTC, amount=0.01000000, open_rate=0.03463000, open_since=closed).
2020-03-09 07:17:49,701 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': sell, 'exchange': 'Binance', 'pair': 'BCH/BTC', 'gain': 'profit', 'limit': 0.034829, 'order_type': 'market', 'amount': 0.01, 'open_rate': 0.03463, 'current_rate': 0.034867, 'profit_amount': 1.3e-06, 'profit_ratio': 0.00373698, 'sell_reason': 'emergency_sell', 'open_date': datetime.datetime(2020, 3, 9, 5, 45, 11, 610951), 'close_date': datetime.datetime(2020, 3, 9, 6, 17, 49, 342573), 'stake_currency': 'BTC', 'fiat_currency': 'SEK'}
2020-03-09 07:17:49,994 - freqtrade.freqtradebot - WARNING - Could not create trailing stoploss order for pair BCH/BTC.
2020-03-09 07:17:49,994 - freqtrade.freqtradebot - WARNING - Unable to sell trade: Attempt to handle closed trade: Trade(id=67, pair=BCH/BTC, amount=0.01000000, open_rate=0.03463000, open_since=closed)
```
So it says it cannot put in stop-loss for "0.03451" as it would sell immediately but then makes a market sell for "0.03482900" that is higher?

*Explain the problem you have encountered*

### Steps to reproduce:

not sure just happens from time to time.
  
### Observed Results:

  * stop-loss on exchange is not placed as it should instead aborts the trade early.

**Comment by xmatthias** [comment]
well letme ask - what's the value of the stoploss? (both trailing and non-trailing).

i've seen this before  when using too tight stoploss - in which case "regular" market movements catch you while cancelling and reapplying the stoploss.

based on the values, the difference in pricing is 1% ... this pair in particular moved 3% this morning within a 5 minute span (i didn't bother to go to minutely levels ...)

then - what's your time offset? you should try to match the values to the chart (best to a timeframe as small as possible - so you can try to feel market movements right at that moment ...).

**Comment by Fredrik81** [comment]
# Optimal stoploss designed for the strategy.
    # This attribute will be overridden if the config file contains "stoploss".
    stoploss = -0.04

    # Trailing stoploss
    trailing_stop = True
    trailing_only_offset_is_reached = True
    trailing_stop_positive = 0.0025
    trailing_stop_positive_offset = 0.0047  # Disabled / not configured

**Comment by xmatthias** [comment]
0.0025 ... that trailing stop will not work - sorry - that's way too tight.

everything < 0.005 is in my opinion - and i personally stay away from everything < 0.01 (but 0.005 should still work though).

It'll produce good backtesting results (read the docs about assumptions made around stoploss to understand why) - but in real conditions it'll encounter the above error way too often, and also the sells will most likely not correspond to the backtesting results - unless you pick pairs VERY carefully.

The error you encounter above will happen whenever the market moves by 0.25% between cancel stoploss, and recreate stoploss - "instant" moves of 1% in crypto are not a real rarity (but i've yet to see a stoploss miss because of a 1% move).

**Comment by Fredrik81** [comment]
Seems like a enhancement for this could be a check-config option that can print out "known" things.
Even validate some strat things perhaps

**Comment by xmatthias** [comment]
i am not 100% certain about this. 
Depending on the pair, a stoploss of 0.0025 may work ... (low volatility - e.g. USDC/USDT).

We've however also got #2696 - so i guess this could be another thing to verify ... ?

**Comment by orehunt** [comment]
if you want to use trailing/stoplosses `<0.005` you should filter the pairlist by spread...that's the main motivation for it..

**Comment by xmatthias** [comment]
I've added the suggestion (verify if stoploss is too tight) to #2696 so we will not forget to add this as strategy validation.
